pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -247,6 +250,14 @@ func localityLbEndpointsFromInstances(instances []*model.ServiceInstance) []endp
 	return out
 }
 
+func connectionId(node string) string {
+	edsClusterMutex.Lock()
+	connectionNumber++
+	c := connectionNumber
+	edsClusterMutex.Unlock()
```

Comment:
A) just use an atomic int for this, so we don't need locks
B) do we really need a monotonically increasing number? Couldn't this problem be solved by just appending a random number to the end of the string? That also avoids needing to synchronize to access the count.

Reply:
we could use source ip + port (though maybe if there is a proxy involved that's not reliable)

also we could require the caller to generate their own nonce

atomic int isn't better than a lock btw, it causes the same serialization under contention

Reply:
Will revisit this in separate PR - this is a quick fix. 

I'm worried about the port - since h2 is multiplexed and some gateways may show up multiplexing further. Long term I thing I would still have one EdsConnection object, with a list of streams going to that endpoint, and remove the lock. Or keep it - we'll iterate over this as we find out what we use for LDS/CDS.

Reply:
Atomics should be better for this case than a lock because atomic add-and-return-value is a hardware instruction (fetch-and-add) vs a mutex lock (which is typically _also_ an FAA) and increment which the compiler likely can't prove is an FAA.